Customer Dashboard

el3ctrician opened the pull request:
**Disclamer** : i am not a professional web developer i barely get by in js :)

### The Issue:

there is a sharing option for creating a dashboard but it was added in the overview page but it allowed only to share one project at time with a link accessiable by everyone. There was no way to customize the shared link, add a password protection or view/delete/edit shared links

### Proposal:

1. Move the share button to be in the project menu since it's only tied to one project
2. Add optional password protection for shared dashboard
3. Add optional slug customization to share a custom link for the dashboard
4. Truly calculated dates of the dashboard from it's period:  a dashboard with last three monthes will always pull the last three monthes instead of fixed dates at the moment of creation
5. Handle password hashing storage in mogo collection, the password implementation is stateless
6. Add admin dashboard section where created dashboard can be viewed, modified and deleted

### Screenshots:

- New location to create dashboards
<img width="1551" height="371" alt="Screenshot from 2025-12-23 15-34-34" src="https://github.com/user-attachments/assets/684d000f-6a95-4c5a-a3c6-4d386a7d7089" />

- New options when creating dashboard
<img width="529" height="584" alt="Screenshot from 2025-12-23 15-35-08" src="https://github.com/user-attachments/assets/d7fa416f-5c47-4b6a-868c-de63d7c22e7d" />

- Auto Sanitize of URL and checks if it's not taken
<img width="529" height="584" alt="Screenshot from 2025-12-23 15-35-49" src="https://github.com/user-attachments/assets/1d886785-4f28-43de-9aae-5349c686f845" />

- New URL generation (if no slug is added then it will still use the ID), passowrd is stored as hashed and salted
<img width="531" height="709" alt="Screenshot from 2025-12-23 15-36-17" src="https://github.com/user-attachments/assets/59f65972-c4fd-458d-90e1-c4c11465227d" />

- Stateless login page accessiable by anyone with the dashboard password
<img width="1922" height="967" alt="Screenshot from 2025-12-23 15-36-44" src="https://github.com/user-attachments/assets/b41adb9a-37b9-4148-a515-5998e022b023" />

- Admin view for created dashboard with edit and delete options
<img width="1342" height="323" alt="Screenshot from 2025-12-23 15-37-42" src="https://github.com/user-attachments/assets/3403268d-6011-4f6d-a94d-3a54c5ca0f00" />

- Edit modal that allows to change all values for dashboard including resetting password

<img width="595" height="754" alt="Screenshot from 2025-12-23 15-37-56" src="https://github.com/user-attachments/assets/d2d971c8-e9ca-40da-984d-8bf6adfeed62" />



### Next steps:

- handle proper translation (need help with other language)
- make some tests by some volunteer users before shipping to make sure that there are no major issues 



PS: Hope that you will appreciate the changes but if you don't like it it's fine i just needed it for my self and i though it's worth sharing with community

el3ctrician committed:
first complete prototype of customer dashboard

faburem commented:
Thanks for the PR, I love the idea and the thoughtful implementation. Please give me some time during the holiday season to review everything in detail before I merge it upstream.

Happy holidays,
Fabian

faburem commented on imports/api/dashboards/server/methods.js:
```diff
@@ -38,12 +68,123 @@ const addDashboard = new ValidatedMethod({
+    }
     const _id = Random.id()
     await Dashboards.insertAsync({
-      _id, projectId, customer, startDate, endDate, resourceId, timeunit, hoursToDays,
```
same as above I would appreciate it if we could keep customer and resourceId as optional parameters

faburem commented on imports/ui/pages/details/components/periodpicker.js:
all these changes are only needed because of the dymic ID? What's the purpose of this change?

faburem commented on imports/ui/pages/overview/components/dashboardModal.js:
```diff
@@ -33,24 +38,104 @@ Template.dashboardModal.helpers({
+
     Meteor.call('addDashboard', {
-      projectId: templateInstance.data.projectId.get(), resourceId: 'all', customer: 'all', timePeriod: templateInstance.$('#period').val(),
+      projectId: templateInstance.data.projectId.get(), timePeriod: period, startDate : startDate, endDate: endDate, password : password,slug:slug
```
see previous comments, please keep resourceId and customer for backwards compatibility if possible

faburem commented:
cool, thanks for your hard work! I will review it and hopefully merge it into the main branch soon.

el3ctrician committed:
added dashboard for user

el3ctrician committed:
added user customer dashboard with filter

el3ctrician committed:
added back customer and resourceID for backward compatibility

el3ctrician committed:
feat: add project-scoped dashboards tab in project settings

- Add Dashboards tab to project edit page (accessible to all project users)
- Refactor dashboardModal to support both create and edit modes
- Create unified projectDashboards component for project and admin views
- Add slug availability checking with current dashboard exclusion
- Add proper date validation for custom period dates
- Add all dashboard-related translations to en.json
- Update admin page to use new unified component
- Add autocomplete=off for password/slug fields to prevent autofill issues
